cpp/src/groupby/sort/groupby.cu

```diff
@@ -299,6 +299,44 @@ void store_result_functor::operator()<aggregation::NUNIQUE>(
   cache.add_result(col_idx, agg, std::move(result));
 };
 
+template <>
+void store_result_functor::operator()<aggregation::NTH_ELEMENT>(
+  std::unique_ptr<aggregation> const& agg)
+{
+  if (cache.has_result(col_idx, agg))
+    return;
+  
+  //FIXME TODO(kn): add include_nulls arg after PR #4159
+  auto count_agg = make_count_aggregation();
+  operator()<aggregation::COUNT>(count_agg);
+  column_view group_sizes = cache.get_result(col_idx, count_agg);
+  
+  auto nth_element_agg =
+    static_cast<experimental::detail::nth_element_aggregation const*>(agg.get());
+
+  //FIXME TODO(kn): remove next 3 lines after PR #4159
+  auto temp_group_sizes = make_numeric_column(
+      cudf::data_type{cudf::experimental::type_to_id<size_type>()},  group_sizes.size(),
+      cudf::UNALLOCATED, stream, mr); 
+  if (nth_element_agg->_include_nulls == include_nulls::YES || !get_grouped_values().has_nulls()) {
+  thrust::adjacent_difference(rmm::exec_policy(stream)->on(stream),
```

Comment:
What's up with indentation? Are you using an auto indent tool?

Reply:
I use vs code. I indent using clang-format on-need basis. I indent when the characters in a line exceeds 80.